### DISCUSSION TITLE
Where is the `Event` object defined?

### DISCUSSION THREAD

**Discussion by LiamPerson** [fabricjs/fabric.js#8643] [Q&A]
I am trying to find the `Event` object such as that required to execute the getPointer function. I can't trace back where this `Event` object is defined and comes from. Where does its data come from and how does that data get parsed?

**Comment by asturur** [comment]
getPointer usually is run over a mouse event object.
https://developer.mozilla.org/en-US/docs/Web/API/MouseEvent

I think we mostly use the clientX and clientY values, but we may change it in the future to get some faster calculations.
Is there any reason why you want to mock the data rather than using a real event?
